tests/test_signer.py

```diff
@@ -0,0 +1,92 @@
+"""
+Tests for signer.py.
+
+A high-level synchronization check between the Account artifacts and Signer module.
+"""
+
+import asyncio
+import sys
+
+import pytest
+from starkware.starknet.services.api.contract_definition import ContractDefinition
+from starkware.starknet.testing.starknet import Starknet
+
+from nile.signer import Signer
+
+SIGNER = Signer(12345678987654321)
+
+
+def get_account_definition():
+    with open("src/nile/artifacts/Account.json", "r") as fp:
+        return ContractDefinition.loads(fp.read())
+
+
+async def send_transaction(
+    signer, account, to, selector_name, calldata, nonce=None, max_fee=0
+):
+    return await send_transactions(
+        signer, account, [(to, selector_name, calldata)], nonce, max_fee
+    )
+
+
+async def send_transactions(signer, account, calls, nonce=None, max_fee=0):
+    if nonce is None:
+        execution_info = await account.get_nonce().call()
+        (nonce,) = execution_info.result
+
+    build_calls = []
+    for call in calls:
+        build_call = list(call)
+        build_call[0] = hex(build_call[0])
+        build_calls.append(build_call)
+
+    (call_array, calldata, sig_r, sig_s) = signer.sign_transaction(
+        hex(account.contract_address), build_calls, nonce, max_fee
+    )
+    return await account.__execute__(call_array, calldata, nonce).invoke(
+        signature=[sig_r, sig_s]
+    )
+
+
+@pytest.fixture(scope="module")
+def event_loop():
+    return asyncio.new_event_loop()
+
+
+@pytest.mark.xfail(
+    sys.version_info >= (3, 9),
+    reason="Issue in cairo-lang. "
+    "See https://github.com/starkware-libs/cairo-lang/issues/27",
+)
```

Comment:
should we add a TODO or notice here to know when to remove it?

Reply:
That's a good idea. I thought the bug StarkWare said they fixed was this frozendict issue, but I'll update the reason^.

Reply:
Actually, maybe they did fix the issue...the test dependency uses starknet-devnet which uses Cairo-v0.8.0. Nile's tests won't reflect the change until the devnet is updated